src/Uno.UI/UI/Xaml/Controls/Border/BorderLayerRenderer.cs

```diff
@@ -1,84 +1,104 @@
 
-namespace Microsoft.UI.Xaml.Shapes;
+namespace Uno.UI.Xaml.Controls;
 
+/// <summary>
+/// Provides Border and Background rendering capabilities to a UI element.
+/// </summary>
 internal partial class BorderLayerRenderer
 {
-	/* The border is defined by the inner and outer rounded-rectangles.
-	 * Each rounded rectangle is composed of 4x lines and 4x 90' arcs.
-	 * ╭─────╮
-	 * │A 2 B│
-	 * │1   3│
-	 * │D 4 C│
-	 * ╰─────╯
-	 * Three factors determine the border: BorderThickness, CornerRadius, and AvailableSize.
-	 * What each part affects:
-	 * - CornerRadius="1A2,2B3,3C4,4D1"
-	 * - BorderThickness="D1A,A2B,B3C,C4D"
-	 * - AvailableSize is used to constrain the radius.
-	 *
-	 * note: technically CornerRadius (together with AvailableSize) also affects the adjacent corner,
-	 * as both points on the same axis will compete for what's available when both add up exceeds
-	 * the available length. (see: h/vRatio in CalculateBorderEllipse)
-	 */
+	private readonly FrameworkElement _owner;
+	private readonly IBorderInfoProvider _borderInfoProvider;
+	private readonly SerialDisposable _borderBrushSubscription = new();
+	private readonly SerialDisposable _backgroundBrushSubscription = new();
 
-	public enum Corner { TopLeft, TopRight, BottomRight, BottomLeft }
+	private BorderLayerState _currentState;
 
-	public static Rect CalculateBorderEllipseBbox(Rect bbox, Corner corner, CornerRadius cr, Thickness bt, bool inner = false)
+	public BorderLayerRenderer(FrameworkElement owner)
 	{
-		var (cr0, hThickness, vThickness, hComplement, vComplement) = corner switch
+		_owner = owner ?? throw new ArgumentNullException(nameof(owner));
+		if (owner is not IBorderInfoProvider borderInfoProvider)
 		{
-			Corner.TopLeft => (cr.TopLeft, bt.Left, bt.Top, cr.TopRight, cr.BottomLeft),
-			Corner.TopRight => (cr.TopRight, bt.Right, bt.Top, cr.TopLeft, cr.BottomRight),
-			Corner.BottomRight => (cr.BottomRight, bt.Right, bt.Bottom, cr.BottomLeft, cr.TopRight),
-			Corner.BottomLeft => (cr.BottomLeft, bt.Left, bt.Bottom, cr.BottomRight, cr.TopLeft),
+			throw new InvalidOperationException("BorderLayerRenderer requires an owner which implements IBorderInfoProvider");
+		}
 
-			_ => throw new ArgumentOutOfRangeException($"Invalid corner: {corner}"),
-		};
+		_borderInfoProvider = borderInfoProvider;
 
-		// there is still a corner to be painted, albeit not rounded.
-		if (cr0 == 0) return AlignCorner(bbox, corner, default);
+		_owner.Loaded += (s, e) => Update();
+		_owner.Unloaded += (s, e) => Clear();
+		_owner.LayoutUpdated += (s, e) => Update();
+	}
 
-		// The ellipse can only grow up to twice the available length.
-		// This is further limited by the ratio between the corner-radius
-		// of that corner and the adjacent corner on the same line.
-		var hRatio = hComplement == 0 ? 1 : (cr0 / (cr0 + hComplement));
-		var vRatio = vComplement == 0 ? 1 : (cr0 / (cr0 + vComplement));
+	/// <summary>
+	/// Updates the border.
+	/// </summary>
+	internal void Update()
```

Comment:
Let's make sure to validate screenshot changes as well.

Reply:
Done and ok!